examples/vision/object_detection_using_vision_transformer.py

```diff
@@ -0,0 +1,605 @@
+"""
+Title: End-to-end Object detection with transformers
+Author: Karan V. Dave
+Date created: 2022/03/27
+Last modified: 2022/03/27
+Description: A simple keras implementation for object detection using transformers.
+"""
+
+
+"""
+## Introduction
+
+ This example is a an implementation of the [Vision Transformer (ViT)]
+ (https://arxiv.org/abs/2010.11929)
+ model by Alexey Dosovitskiy et al. for object detection,
+ it is demonstrated on the Caltech 101 dataset for detecting airplane in an image.
+
+ This example requires TensorFlow 2.4 or higher, and
+ [TensorFlow Addons](https://www.tensorflow.org/addons/overview),
+ which can be installed using the following command:
+```
+ pip install -U tensorflow-addons
+```
+"""
+
+"""
+## Imports and setup
+"""
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import tensorflow_addons as tfa
+import matplotlib.pyplot as plt
+import numpy as np
+import cv2
+import os
+import scipy.io
+
+from tensorflow.keras.preprocessing.image import img_to_array
+from tensorflow.keras.preprocessing.image import load_img
+
+import tensorflow as tf
+import datetime
+
+"""
+ To run the TensorBoard from a Jupyter notebook,
+ load the TensorBoard notebook extension inside notebook
+```
+%load_ext tensorboard
+```
+"""
+
+"""
+## Prepare dataset
+"""
+
+# Dataset
+# http://www.vision.caltech.edu/Image_Datasets/Caltech101/
+# Download the images '101_ObjectCategories.tar.gz' and
+# annotations 'Annotations.tar' available seperately
+# Extract images (using airplanes here) to images folder and
+# all annotations to another folder.
+
+# Path to images and annotations
+PATH_IMAGES = "C:/Users/DAVE/Desktop/Object detection with transformers/airplanes/"
```

Comment:
Rather than hard coding paths on your own disk, write Python code that downloads the file (e.g. using `keras.utils.get_file`, uncompressed the data, etc.